cli/conf.py

```diff
@@ -105,6 +123,81 @@ def parse_args(cls, module_name, config, args=None):
 
         return res_args
 
+
+class SpecLoader(object):
+    """
+    Responsible to load spec files.
+    """
+    SPEC_EXTENSION = '.spec'
+    LOOKUP_DICT = {
+        '__DEFAULT__': "default"
+    }
+
+    TRIM_PARAMS = ['default', 'required']
+
+    @classmethod
+    def load(cls, module_name, config):
+        """
+        Loads the spec file as dict.
+        """
+        clg_options = cls._get_specs(module_name, config)
+        global_options, subcommands_options = cls._get_options(clg_options)
+        return clg_options, global_options, subcommands_options
+
+    @classmethod
+    def _get_options(cls, spec_dict, lookup=False, trim=False):
+        global_options = {}
+        subcommand_options = {}
+        cls.__get_options(global_options, spec_dict, lookup, trim)
+        cls.__get_groups_options(global_options, spec_dict, lookup, trim)
+        # collect sub parser global_options
+        if 'subparsers' in spec_dict:
+            for subparser_name, params in spec_dict['subparsers'].iteritems():
+                subcommand_options[subparser_name], _ = cls._get_options(
+                    params, lookup=True, trim=True)
+
+        return global_options, subcommand_options
+
+    @classmethod
+    def __get_groups_options(cls, options, spec_dict, lookup, trim):
+        if 'groups' in spec_dict:
+            for group in spec_dict['groups']:
+                cls.__get_options(options, group, lookup, trim)
+
+    @classmethod
+    def __get_options(cls, options, spec_dict, lookup, trim):
+        if 'options' in spec_dict:
+            for opt_name, opt_params in spec_dict['options'].iteritems():
+                if lookup:
+                    cls.lookup_option(opt_name, opt_params)
+
+                # get a parameters copy with all the keys.
+                options[opt_name] = dict(opt_params)
+
+                if trim:
+                    cls.trim_option(opt_name, opt_params)
+
+    @classmethod
+    def lookup_option(cls, opt_name, opt_params):
+        # first check __*__ pattern
+        for opt_param_name, param_value in opt_params.iteritems():
+            for lookup_string, lookup_key in cls.LOOKUP_DICT.iteritems():
+                if lookup_string in str(param_value):
+                    opt_params[opt_param_name] = \
+                        param_value.replace(lookup_string, str(opt_params.get(
+                            lookup_key, lookup_string)))
+
+        # insert default value into help.
+        if 'help' in opt_params and 'default' in opt_params:
+            opt_params['help'] += " | Default value: {}".format(
+                opt_params['default'])
+
+    @classmethod
+    def trim_option(cls, opt_name, opt_params):
```

Comment:
Guessing: "removes clg attributes (default and required) from clg argument."
If so, please rename "opt_name" and "opt_params". If not, please provide a proper docstring